src/main/java/frc/robot/subsystems/funnel/FunnelCommandsBuilder.java

```diff
@@ -0,0 +1,89 @@
+package frc.robot.subsystems.funnel;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.FunctionalCommand;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+
+import java.util.function.DoubleSupplier;
+
+public class FunnelCommandsBuilder {
+
+    private final Funnel funnel;
+
+    public FunnelCommandsBuilder(Funnel funnel){
+        this.funnel = funnel;
+    }
+
+    public Command setBigFunnelPower(double power){
+        return new FunctionalCommand(
+                () -> funnel.setBigFunnelPower(power),
+                () -> {},
+                interrupted -> funnel.stopBigFunnel(),
+                () -> false,
+                funnel
+        ).withName("Set big funnel power: " + power);
+    }
+
+    public Command setMiddleFunnelPower(double power){
+        return new FunctionalCommand(
+                () -> funnel.setMiddleFunnelPower(power),
+                () -> {},
+                interrupted -> funnel.stopMiddleFunnel(),
+                () -> false,
+                funnel
+        ).withName("Set middle funnel power: " + power);
+    }
+
+    public Command setBigFunnelPower(DoubleSupplier power) {
+        return new FunctionalCommand(
+                () -> {},
+                () -> funnel.setBigFunnelPower(power.getAsDouble()),
+                interrupted -> funnel.stopBigFunnel(),
+                () -> false,
+                funnel
+        ).withName("Set big funnel power by supplier");
+    }
+
+    public Command setMiddleFunnelPower(DoubleSupplier power) {
+        return new FunctionalCommand(
+                () -> {},
+                () -> funnel.setMiddleFunnelPower(power.getAsDouble()),
+                interrupted -> funnel.stopMiddleFunnel(),
+                () -> false,
+                funnel
+        ).withName("Set middle funnel power by supplier");
+    }
+
+
+    public Command rollBigFunnelRotations(Rotation2d rotations, double power) {
```

Comment:
function name "rollBigFunnelRotations" is overly verbose. consider removing type of first function parameter from it's name, since it's already part of the function's signature.
```suggestion
    public Command rollBigFunnel(Rotation2d rotations, double power) {
```